File: STRINGS/1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDepth = function(s) {
    let st = [];
    let maxCounter = 0;
    
    for (let i = 0; i < s.length; i++) {
        if (s[i] === '(') {
            st.push(i);
        } else if (s[i] === ')') {
            st.pop();
        }
        maxCounter = Math.max(maxCounter, st.length);
    }
    
    return maxCounter;
};

File: STRINGS/1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.rb
def max_depth(s)
    st = []
    max_counter = 0
    
    s.each_char do |char|
        if char == '('
            st.push(1)
        elsif char == ')'
            st.pop
        end
        max_counter = [max_counter, st.size].max
    end
    
    max_counter
end
